### PR TITLE
Remove Err and Log definitions

### PR DESCRIPTION
**Pull request by orenbm**
Note: it is best to review this PR through the commits

#### Remove Err and Log definitions from top of classes

Each time we used an error class, or a log message class,
we would define its module at the top of the file. This was done
so when you call the class it would be concise without the whole path.

This approach led to issues where because we used memoization to init these
constants, they would sometimes mismatch what is written. For example,
if we had `Err ||= Errors::Authentication::AuthnAzure` in the top of the class
then sometimes the actual value of `Err` was `Errors::Authentication` because it was
already initialized with that value and the memoization meant that we don't initialize
it again.

A solution for that was to remove the memoization whenever this happened but then we had
another issue, where we got "already initialized constant" warnings. This happens because
now we initialize the same constant over and over again (e.g Err = Errors::Authentication::AuthnAzure).

Adding to the above the advantage of the explicit readability of calling error classes with their
full path led to the conclusion that we should remove the definition of `Err` and `Log` from the
top of classes, and call the full path each time.

#### Remove possible errors raised comments

We currently add a comment to each command class where we
specify the possible errors that are raised by this class.

This comment is hard-coded and it is the responsibility of the
developers to maintain it. The problem with this comment is that
it is not generated from anywhere and is not verified by code.

This leads us to a lot of cases where we move errors and forget to
update the comment so the comments are not accurate. Also, because
it is not verified then some classes have it and some classes do not.

It is better to not have any data at all than to have data that is incorrect,
thus we should remove these comments until they are forced by code of the
CommandClass interface.

### What ticket does this PR close?
Connected to #972 

#### Change log
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] The changes in this PR do not require additional tests

#### Documentation
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: app/domain/authentication/authenticate.rb [hunk 1]
```diff
@@ -45,11 +41,11 @@ def authenticator
     end
 
     def validate_authenticator_exists
-      raise Err::AuthenticatorNotFound, authenticator_name unless authenticator
+      raise Errors::Authentication::AuthenticatorNotFound, authenticator_name unless authenticator
```
Comment (jonahx) [inline app/domain/authentication/authenticate.rb:44]: I like this change, but I don't like these long lines.  How about:

```ruby
      raise(
        Errors::Authentication::AuthenticatorNotFound,
        authenticator_name
      ) unless authenticator
```
whenever the long gets too long?

Comment (orenbm) [reply app/domain/authentication/authenticate.rb:44]: sure, i'll do it.

File: app/domain/authentication/authenticate.rb [hunk 2]
```diff
@@ -4,7 +4,6 @@
 
 module Authentication
 
-  Err ||= Errors::Authentication
```
Comment (sigalsax) [inline app/domain/authentication/authenticate.rb:7]: Found the following as part of the design for authn-azure and azuthn-status

```
    Err = Errors::Authentication::AuthnAzure
```

```
    Err = Errors::Authentication::Authn<Type>
```

Should we update those too?

Comment (sigalsax) [reply app/domain/authentication/authenticate.rb:7]: along with

```
 # Possible Errors Raised:
```

File: app/domain/logs.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ module LogMessages
 
   module Conjur
 
-    PrimarySchema = Util::TrackableErrorClass.new(
+    PrimarySchema = ::Util::TrackableLogMessageClass.new(
```
Comment (sigalsax) [inline app/domain/logs.rb:7]: why was :: added? From this commit, it doesn't seem that there was a change in directory

Comment (orenbm) [reply app/domain/logs.rb:7]: i rooted the module so there is no confusion with other Util modules

File: app/domain/authentication/authenticate.rb (hunk 2 above)
Comment (orenbm) [reply app/domain/authentication/authenticate.rb:7]: Nah, i don't think we need to update old designs when we perform refactors

**Pull request merged**